tests/components/broadlink/test_init.py

```diff
@@ -0,0 +1,109 @@
+"""The tests for the broadlink component."""
+from datetime import timedelta
+from base64 import b64decode
+from unittest.mock import MagicMock, patch
+
+import pytest
+import voluptuous as vol
+
+from homeassistant.util.dt import utcnow
+from homeassistant.components.broadlink import async_setup_service
+from homeassistant.components.broadlink.const import (
+    DOMAIN, SERVICE_LEARN, SERVICE_SEND)
+
+DUMMY_IR_PACKET = ("JgBGAJKVETkRORA6ERQRFBEUERQRFBE5ETkQOhAVEBUQFREUEBUQ"
+                   "OhEUERQRORE5EBURFBA6EBUQOhE5EBUQFRA6EDoRFBEADQUAAA==")
+DUMMY_HOST = "192.168.0.2"
+
+
+@pytest.fixture(autouse=True)
+def dummy_broadlink():
+    """Mock broadlink module so we don't have that dependency on tests."""
+    broadlink = MagicMock()
+    with patch.dict('sys.modules', {
+            'broadlink': broadlink,
+    }):
+        yield broadlink
+
+
+async def test_send(hass):
+    """Test send service."""
+    mock_device = MagicMock()
+    mock_device.send_data.return_value = None
+
+    async_setup_service(hass, DUMMY_HOST, mock_device)
+    await hass.async_block_till_done()
+
+    await hass.services.async_call(DOMAIN, SERVICE_SEND, {
+        "host": DUMMY_HOST,
+        "packet": (DUMMY_IR_PACKET)
+    })
+    await hass.async_block_till_done()
+
+    mock_device.send_data.assert_called_with(b64decode(DUMMY_IR_PACKET))
+
+
+async def test_learn(hass):
+    """Test learn service."""
+    mock_device = MagicMock()
+    mock_device.enter_learning.return_value = None
+    mock_device.check_data.return_value = b64decode(DUMMY_IR_PACKET)
+
+    with patch.object(hass.components.persistent_notification,
+                      'async_create') as mock_create:
+
+        async_setup_service(hass, DUMMY_HOST, mock_device)
+        await hass.async_block_till_done()
+
+        await hass.services.async_call(DOMAIN, SERVICE_LEARN, {
+            "host": DUMMY_HOST,
+        })
+        await hass.async_block_till_done()
+
+        mock_device.enter_learning.assert_called_with()
```

Comment:
We try to avoid using assertion methods or attributes. They risk passing tests that should fail if an assertion method is misspelled.

Use:
```py
assert mock_device.enter_learning.call_args.call_count == 1
assert mock_device.enter_learning.call_args == mock.call(...)
```